addons/libpthread/pthread-internal.h

```diff
@@ -39,7 +49,10 @@ typedef union pthread_mutex_t {
 } pthread_mutex_t;
 
 typedef union pthread_cond_t {
-    condvar_t cond;
+    struct {
+        condvar_t cond;
+        const pthread_condattr_t *attr;
```

Comment:
Storing a pointer like this is almost certainly not the correct way to do this.

If the attribute is allocated on the stack and goes out of scope before the condvar does, this will probably break. I suspect that that is not exactly an uncommon case to have happen (i.e, some function sets up/inits things for a statically allocated condvar in .bss/.data, but the attr is made on the stack in that function since it shouldn't need to be around after the condvar is created).

Reply:
You probably just want to copy the clockid into the structure (or copy the whole condattr into it if you want to be really safe).

Reply:
Also, since this will probably change structure a bit of other parts of this, I'll hold off on reviewing the rest of the PR until this is resolved.